tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -513,6 +520,12 @@ Status FragmentConsolidator::copy_array(
     // READ
     RETURN_NOT_OK(query_r->submit());
 
+    // If Consolidation cannot make any progress, throw.
+    if (buffer_sizes->at(0) == 0) {
```

Comment:
Would it make sense to check this condition in the caller instead?

Reply:
What are the semantics of `buffer_sizes`? I don't know which buffer `->at(0)` this is referring to -- is this going to work correctly if the first buffer is for an array of empty strings?

Reply:
Thanks for the clarification.